app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+              ->setName('fullname')
+              ->setLabel('ライター名')
+              ->setSortable(true)
+              ->setSorting(Grid::SORT_ASC)
+              ->addFilter(
+                (new SelectFilterConfig())
+                  ->setDefaultValue(null)
+                  ->setOperator(FilterConfig::OPERATOR_EQ)
+                  ->setTemplate('webapp.components.nayjest_columns_filter')
+                  ->setFilteringFunc(function($val, DbalDataProvider $dap) {
+                      return $dap->getBuilder()->andWhere("u.id = $val")->addGroupBy('p.id');
+                  })
+              )
+                ->setCallback(function($val, ObjectDataRow $obj) {
+                    $user_id = $obj->getCellValue('user_id');
+                    if($this->currentUser->is('admin')) {
+                        return '<a href="' . action('Webapp\UserController@getEdit', $user_id) . '">' . $val . '</a>' ;
+                    }
+                    return $val;
+                }),
+            (new FieldConfig)
+              ->setName('keyword')
+              ->setLabel('キーワード')
+              ->addFilter(
+                (new SelectFilterConfig())
+                  ->setDefaultValue(null)
+                  ->setName('keyword_fil')
+                  ->setTemplate('webapp.components.nayjest_columns_filter')
+                  ->setOperator(FilterConfig::OPERATOR_EQ)
+                  ->setFilteringFunc(function($val, DbalDataProvider $dap) {
+                      return $dap->getBuilder()->andWhere("k.id = $val")->addGroupBy('p.id');
+                  })
+              )
+              ->setCallback(function($val, ObjectDataRow $obj) {
+                  return $obj->getCellValue('type') == 1 ? '<a href="' . url('/webapp/post') . '?keyword=' . urlencode($val) . '">' . $val . '</a>' : '';
+              }),
+            (new FieldConfig)
+              ->setName('published_status')
+              ->setLabel('ステータス')
+              ->setSortable(true)
+              ->setSorting(Grid::SORT_ASC)
+              ->setCallback(function($val) {
+                  return $val == 1 ? $val = '<span class="label label-success fa fa-check"><span class="tooltip action-top">公開済み</span></span>' : $val = '<span class="label label-default fa fa-minus-circle"><span class="tooltip action-top">下書き</span></span>';
+              })
+              ->addFilter(
+                  (new SelectFilterConfig())
+                      ->setFilteringFunc(function($val, DbalDataProvider $dap) {
+                          return $dap->getBuilder()->andWhere("p.published_status = $val")->addGroupBy('p.id');
+                      })
+                    ->setOptions(['1' => '公開済み', '0' => '下書き'])
+              ),
+            (new FieldConfig)
+              ->setName('name')
+              ->setLabel('記事リンク')
+              ->addFilter(
+                (new SelectFilterConfig)
+                  ->setDefaultValue(null)
+                  ->setName('site_id')
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "site_id" 13 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)